lib/collection/src/shard/proxy_shard.rs

```diff
@@ -70,27 +114,39 @@ impl ShardOperation for &ProxyShard {
     ) -> CollectionResult<UpdateResult> {
         let local_shard = &self.wrapped_shard;
         let estimate_effect = operation.estimate_effect_area();
-        match estimate_effect {
-            OperationEffectArea::Empty => {}
-            OperationEffectArea::Points(points) => {
-                self.check_changed_points_limit(points.len()).await?;
-                let mut changed_points_guard = self.changed_points.write().await;
-                for point in points {
-                    changed_points_guard.insert(point);
-                }
-            }
+        let points_operation_effect: PointsOperationEffect = match estimate_effect {
+            OperationEffectArea::Empty => PointsOperationEffect::Empty,
+            OperationEffectArea::Points(points) => PointsOperationEffect::Some(points),
```

Comment:
We should a comment explaining that we do not track the number of points added on purpose.